### ISSUE TITLE
/bin/garn: line 2: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8)

### ISSUE THREAD

**Issue by khaledez** [garnix-io/garn#403]
Hello,
I'm trying garn on macOS, I'm seeing this warning every time I run garn command

```
$HOME/.nix-profile/bin/garn: line 2: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8): No such file or directory
```

This is the result of running `locale`
```
LANG=""
LC_COLLATE="C"
LC_CTYPE="UTF-8"
LC_MESSAGES="C"
LC_MONETARY="C"
LC_NUMERIC="C"
LC_TIME="C"
LC_ALL=
```

**Comment by jkarni** [comment]
Does this happen for `garn build` and `garn check` as well as `garn enter`? And do you see the logs the same number of times?

**Comment by khaledez** [comment]
> Does this happen for `garn build` and `garn check` as well as `garn enter`? And do you see the logs the same number of times?

Yes, it happens with all commands.

**Comment by khaledez** [comment]
executing `head -n 5 $HOME/.nix-profile/bin/garn` gives this result:
```sh
#! /nix/store/zzpm4317hn2y29rm46krsasaww9wxb1k-bash-5.2-p15/bin/bash -e
export LC_ALL='C.UTF-8'
PATH=${PATH:+':'$PATH':'}
PATH=${PATH/':''/nix/store/553wiycn7i2xbfykssl97smg1pwlgfg0-cabal2json/bin'':'/':'}
PATH='/nix/store/553wiycn7i2xbfykssl97smg1pwlgfg0-cabal2json/bin'$PATH
```

**Comment by alexdavid** [comment]
It looks like this environment variable was changed from `LANG` to `LC_ALL` in ae69820356d91edf255d8f5e05b1ee2dee778886. `LANG` was originally introduced in 3e52fd00e7965f47c109e36d3611c4d37305e731.

I've removed the variables completely in 665f623c133b4923bf2d3e98d3ccea5abd0b4975, but haven't opened a PR yet since I'm not sure of the ramifications. In my limited testing though I am still able to `nix profile install` garn and everything seems to work well both inside and outside of the garn repo -- the environment variable isn't set on my system. `LANG` is set, but calling garn with `LANG='' garn` also seems to have no noticeable impact on it. Can anyone see any reason not to remove these?
